File: app/src/main/java/com/taotao/healthtracker/data/entity/HealthRecord.kt
package com.taotao.healthtracker.data.entity

import androidx.room.*

@Entity(tableName = "user_profiles")
data class UserProfile(
    @PrimaryKey val id: Int = 1,
    val name: String = "Me",
    val birthYear: Int = 1990,
    val birthMonth: Int = 1,
    val birthDay: Int = 1,
    val height: Float = 175f,
    val language: String = "zh",
    val insightLanguage: String = "zh" // Independent lang for the bottom card
)

@Entity(tableName = "almanac_data")
data class AlmanacData(
    @PrimaryKey val date: String, // YYYY-MM-DD
    val yi: String,
    val ji: String,
    val lunarDate: String
)

@Entity(tableName = "health_records")
data class HealthRecord(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val userId: Int,
    val date: String,
    val sbp: Int?,
    val dbp: Int?,
    val hr: Int?,
    val weight: Float?
)

File: Healthtracker/Healthtracker/HealthRecord.swift
import SwiftData
import Foundation

@Model
final class HealthRecord {
    var id: UUID
    var userId: Int
    var date: Date
    var sbp: Int?
    var dbp: Int?
    var hr: Int?
    var weight: Double?
    
    init(
        id: UUID = UUID(),
        userId: Int,
        date: Date,
        sbp: Int? = nil,
        dbp: Int? = nil,
        hr: Int? = nil,
        weight: Double? = nil
    ) {
        self.id = id
        self.userId = userId
        self.date = date
        self.sbp = sbp
        self.dbp = dbp
        self.hr = hr
        self.weight = weight
    }
}
